src/Utilities/Algorithm.hpp

```diff
@@ -0,0 +1,68 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <numeric>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Get the indices in the container `t` of the `K` elements that satisfy
+ * `op` after sorting with respect to `op`.
+ *
+ * Specifically, if `op` is `[](const lhs, const rhs) { return lhs < rhs; }`
+ * then the result will be the indices of the `K` smallest elements in `t`. If
+ * `op` is `[](const lhs, const rhs) { return lhs > rhs; }` then the result will
+ * be the indices of the `K` largest elements in `t`.
+ *
+ * \warning The indices are not sorted by `op` when returned. Use
+ * `sorted_indices_of`.
+ */
+template <size_t K, typename T, typename Comparator>
+auto indices_of(const T& t, Comparator op) noexcept -> std::array<size_t, K> {
```

Comment:
I do not like the name.  It does not give any suggestion of *what* it gives the indices of.  Perhaps `indices_of_smallest`?

Reply:
The flexibility to specify the comparator means the "smallest" indices could actually correspond to bigger values, which I worry may lead to confusing code, e.g.
```
const auto indices_of_largest_eigenvalues = indices_of_smallest(eigenvalues, [](...){ ... > ... });
```

I could be convinced that this is okay, especially if the documentation were extremely clear.

Reply:
Yes, but it's not called `partial_sort_smallest`................

Reply:
I'm not sure what you want changed. I'm happy to change the name, I just don't know what to. I think it's reasonable to have the documentation focus on a specific example, e.g. compare less.